interp/constrextern.ml

```diff
@@ -63,6 +63,31 @@ let print_universes = Detyping.print_universes
 (* This suppresses printing of primitive tokens (e.g. numeral) and notations *)
 let print_no_symbol = ref false
 
+(* This tells to skip types if a variable has this type by default *)
+let print_use_implicit_types = ref true
+
+let () =
+  let open Goptions in
+  declare_bool_option
+    { optdepr  = false;
+      optname  = "use of implicit types in printing";
+      optkey   = ["Printing";"Use";"Implicit";"Types"];
+      optread  = (fun () -> !print_use_implicit_types);
+      optwrite = (:=) print_use_implicit_types }
+
+(**********************************************************************)
+
+let skip_if_reserved_type na t =
+  if !Flags.raw_print || not !print_use_implicit_types then t else
+  match na with
+  | Anonymous -> t
+  | Name id ->
+    try
+      let pat = Reserve.find_reserved_type id in
+      let _ = match_notation_constr false t ([],pat) in
+      DAst.make ?loc:t.CAst.loc @@ GHole (Evar_kinds.InternalHole, IntroAnonymous, None)
+    with Not_found -> t
```

Comment:
Doesn't match_notation_constr raise No_match?

Reply:
(if this needs to be fixed please ensure there is a test for it)

Reply:
You're entirely correct. Exception and test added.